src/libstore/builtins/buildenv.hh

```diff
@@ -12,6 +12,66 @@ struct Package {
     Package(const Path & path, bool active, int priority) : path{path}, active{active}, priority{priority} {}
 };
 
+class BuildEnvFileConflictError : public Error
+{
+public:
+    const Path originalFile;
+    const Path conflictingFile;
+    int priority;
+
+    BuildEnvFileConflictError(
+        const Path originalFile,
+        const Path conflictingFile,
+        int priority
+    )
+        : Error(
+            "Unable to build profile. There is a conflict for the following files:\n"
+            "\n"
+            "  %1%\n"
+            "  %2%",
+            originalFile,
+            conflictingFile
+        )
```

Comment:
Side note, we really have a separate virtual method for *rendering* errors so we aren't storing both the structured data and a big long preformatted string. This is an awkard redundancy that just tempts people into overusing the plain `Error` type.

Reply:
Side note split into issue #7865.

Reply:
Hmm, I see I could override `what`, but quite a few places use `.msg()` instead of `what`. `msg` doesn't seem virtual.

I looked around in the codebase, but it seems there aren't that many custom (structured) errors, as they all use `MakeError`.

What would be the right way to implement 'lazy' error messages?

Reply:
@bobvanderlinden to be clear, I didn't mean this to block this PR. But if you would like to take a stab at this issue, I am happy to provide guidance over there.